clib/ivar.c

```diff
@@ -122,15 +137,27 @@ void ivar_get_with_size( void *var, struct ivar iv, int size )
     log_3("ivar_get_with_size %p %p %d - leave\n", var, &iv, size);
 }
 
-struct array * ivar_get_array( struct array *var, struct ivar iv )
+struct array * ivar_get_array( struct array *var, struct ivar iv, void* vcf )
+{
+    struct array *ptr;
+    log_2("ivar_get_array %p %p - enter\n", var, &iv);
+    ivar_get_helper(iv.internals);
+    ptr = (struct array*)iv.internals->data;
```

Comment:
malloc(): I still don't get it. malloc() is declared to return a 'void*'. Can you really put a 'void*' in a variable with type 'struct array*'? I thought there was something akin to a subtype ordering.

Comments: ok, /// makes sense.

ivar.c: I copy-pasted and wanted to change as little as possible; this is the way the non-shallow original was written. Should've changed the log, though.

Reply:
A pointer conversion to and from void * is implicit in C so it will "fit".

I haven't looked at how the Clang error detection for size mismatches works under the hood but I know that if you cast the result, you disable the warnings (since the cast indicates you want a specific type, even if it's the wrong one).

Reply:
Ok. This was new to me!